CMakeLists.txt

```diff
@@ -63,6 +63,8 @@ endif()
 ament_export_include_directories(include)
 ament_export_libraries(${PROJECT_NAME})
 
+ament_export_dependencies(roscpp)
```

Comment:
This isn't going to work the way intended. `roscpp` is a ROS 1 package and should be found using the custom CMake function used above (`find_ros1_package` which doesn't rely on CMake config files but `pkg-config` instead). If you want this to be exposed in downstream package it would need to be done in an extra file by invoking `find_ros1_package`.

Reply:
Hi @dirk-thomas, our problem is the following: we link against the ros1_bridge and on Linux everything works fine, whereas on Mac the linker cannot find `roscpp` symbols.
We could solve this by finding `roscpp` ourselves, as you said, but for us it is only a transitive dependency for the bridge, therefore we thought that this would be the best solution: with this change, the `roscpp` dependency is exported and everything works as expected.

Reply:
I understand your use case but this is still the wrong approach. As soon as a `roscpp` package is created in ROS 2 e.g. to provide an API shim the current approach is going to fail. The `ros1_bridge` package needs to invoke `find_ros1_package` in an extra file instead.

Reply:
Ok, we will then use the `find_ros1_package` macro in our package.

Reply:
I am not saying you have call it in your package. The logic can be placed in a CMake extra file in `ros2_bridge` in order to avoid having to repeat it in multiple downstream packages.